### ISSUE TITLE
password decrypted to early

### ISSUE THREAD

**Issue by L0ric0** [carnager/rofi-pass#212]
when opening the action menu with `Alt+a` the password is decrypted and then the dialog for the next step is shown. After choosing an action the password is decrypted again. The first decryption is not needed and is annoying for everyone who has to touch their hardware token to decrypt a password.

**Comment by hcgrove** [comment]
Den 05.09.2021 kl. 18.51 skrev L0ric0:
>
> when opening the action menu with |Alt+a| the password is decrypted
> and then the dialog for the next step is shown. After choosing an
> action the password is decrypted again. The first decryption is not
> needed and is annoying for everyone who has to touch their hardware
> token to decrypt a password.
>

What?

Your configuration must be a bit strange, here Alt+a does nothing
related to rofi-pass. I think you must have bound that in a bad way,
which command does Alt+a invoke on your system?
Which password is decrypted? What actions can you choose?

.Henrik

**Comment by L0ric0** [comment]
I'm using the AUR package `rofi-pass-ydotool-git` which uses the ydotool branch. The keybindings are the ones supplied by the package.

The `Alt+a` keybind invokes the `action_menu` and there the options to choose are:
1. move password file
2. copy password file
3. delete password file
4. edit password file
5. generate new password

I'm using a hardware token that has to be touched each time the secret key is used and when pressing `Alt+a` to open the `action_menu` it prompts me to touch it. This happens before the menu is shown. Then when choosing to edit the password I'm again prompted to touch the hardware token to decrypt the password and after editing again.

**Comment by carnager** [comment]
Yup, this is annoying. mainMenu function should be cleaned up quite a bit, so unneeded operations are not executed.
